### DISCUSSION TITLE
Inquiry about audio switching feature in Plyr with HLS (multiple audio tracks for video) and possible implementation

### DISCUSSION THREAD

**Discussion by sinazahed** [sampotts/plyr#2705] [Q&A]
Hello Plyr team and community!

First of all, I would like to express my gratitude for the Plyr library. It has been a fantastic tool for implementing video and audio playback on my website, and I truly appreciate the effort and dedication put into developing and maintaining this open-source project.

I have been exploring the features and capabilities of Plyr, particularly with HLS (HTTP Live Streaming) files. I'm looking to provide users with the ability to switch between multiple audio tracks within a video. However, I haven't been able to find a built-in audio switching feature within Plyr specifically for HLS video streams.

I would like to inquire if Plyr supports audio track switching for HLS streams out-of-the-box. If so, I would appreciate any guidance or examples on how to implement this feature.

If audio switching is not currently supported in Plyr for HLS streams, I would like to seek advice on the best approach to achieve this functionality. Are there any recommended plugins, extensions, or customizations that can be integrated with Plyr to enable audio track switching?

I understand that this may involve additional libraries or APIs specialized in working with HLS streams, such as hls.js. If so, I would be grateful for any suggestions or pointers on how to integrate Plyr with such libraries to accomplish the desired audio switching functionality.

Thank you once again for your exceptional work on Plyr, and I look forward to hearing your insights and recommendations on this matter.

Best regards,

**Comment by Deep-Bag** [comment]
Did you find a way to switch audio tracks? @sinazahed

**Reply by ahoura** [reply]
Hey, 

I was looking for the same thing and ended up building something myself, note that I handle controls/GUI myself so I can tell you how to do the js side of things and youll need to add GUI components yourself.

```
var hls = new Hls();
var audioTracks = [];
hls.loadSource('https://cdn.bitmovin.com/content/assets/sintel/hls/playlist.m3u8');
hls.attachMedia(video);
hls.on(Hls.Events.AUDIO_TRACKS_UPDATED, () => {
          if (hls.audioTracks.length > 1) {
            audioTracks = hls.audioTracks;
          }
          // RENDER YOUR GUI HERE
});

switchAudioTrack(index) {
        // Set the audio track via hls.js.
        hls.audioTrack = parseInt(index, 10);
        this.currentAudioTrack = hls.audioTrack;
    }
```
This will store audio tracks in audioTracks array **after** the event `Hls.Events.MANIFEST_PARSED` has been called.
